command-ssh: include ucred.h

In order to use ucred() when HAVE_SO_PEERCRED is defined, ucred.h needs
to be included on Solaris.

https://bugs.gnupg.org/gnupg/issue2981

--- a/agent/command-ssh.c
+++ b/agent/command-ssh.c
@@ -40,6 +40,9 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <assert.h>
+#ifdef HAVE_UCRED_H
+#include <ucred.h>
+#endif
 
 #include "agent.h"
 
